### PR TITLE
Module/rds snapshot explorer

### PR DESCRIPTION
**Pull request by alexanderinsa**

### CODE REVIEW

**Review by SpenGietz — CHANGES_REQUESTED** [review]

File: modules/rds_snapshot_explorer/main.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+def restore_instance_from_snapshot(client, snapshot, print):
+    waiter = client.get_waiter('db_snapshot_available')
+    waiter.wait(
+        DBSnapshotIdentifier=snapshot['DBSnapshotIdentifier'],
+        WaiterConfig=WAIT_CONFIG,
+    )
+    try:
+        response = client.restore_db_instance_from_db_snapshot(
+            DBInstanceIdentifier=snapshot['DBSnapshotIdentifier'],
```
[the remark this answers is not in the document]
Comment (alexanderinsa) [reply modules/rds_snapshot_explorer/main.py]: This is intentional. My thinking was that given an instance, you create a snapshot identifier based on the existing instance's identifier and concatenate '-copy' to the snapshot identifier. Identifier's have to be unique for the particular resource, so you have a snapshot and instance identifier be exactly the same. I could change the '-copy' to a random string of characters to reduce the possibility of collisions further though.

File: modules/rds_snapshot_explorer/main.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+def create_snapshot_from_instance(client, instance, print):
+    waiter = client.get_waiter('db_instance_available')
+    waiter.wait(
+        DBInstanceIdentifier=instance['DBInstanceIdentifier'],
+        WaiterConfig=WAIT_CONFIG,
+    )
+    try:
+        response = client.create_db_snapshot(
+            DBSnapshotIdentifier=instance['DBInstanceIdentifier'] + '-copy',
```
[the remark this answers is not in the document]
Comment (alexanderinsa) [reply modules/rds_snapshot_explorer/main.py]: Yes

File: modules/rds_snapshot_explorer/main.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        return out
+    except ClientError as error:
+        print('    ' + error.response['Error']['Code'])
+        return []
+
+
+def summary(data, pacu_main):
+    if 'fail' in data:
+        return data['fail']
+    return '  {} Instance(s) Copies Launched'.format(data['instances'])
```
[the remark this answers is not in the document]
Comment (alexanderinsa) [reply modules/rds_snapshot_explorer/main.py]: I will implement this.

File: modules/rds__explore_snapshots/main.py [hunk 4]
```diff
@@ -6,11 +6,11 @@
 from botocore.exceptions import ClientError
 
 module_info = {
-    'name': 'rds_snapshot_explorer',
+    'name': 'rds__explorer_snapshots',
```
Comment (SpenGietz) [inline modules/rds__explore_snapshots/main.py]: "explore"

File: modules/rds__explore_snapshots/main.py [hunk 5]
```diff
@@ -6,11 +6,11 @@
...
     'author': 'Alexander Morgenstern alexander.morgenstern@rhinosecuritylabs.com',
-    'category': 'post-exploitation',
+    'category': 'EXPLOIT',
```
Comment (SpenGietz) [inline modules/rds__explore_snapshots/main.py]: Probably throw this one under exfil

File: modules/rds__explore_snapshots/main.py [hunk 6]
```diff
@@ -63,10 +63,12 @@ def cleanup(pacu):
     success = True
     for instance in data['Instances']:
         client = pacu.get_boto3_client('rds', data['Instances'][instance]['AvailabilityZone'][:-1])
-        delete_instance(client, instance, pacu.print)
+        if not delete_instance(client, instance, pacu.print):
```
Comment (SpenGietz) [inline modules/rds__explore_snapshots/main.py]: Will this encounter errors/handle them? Same question for line 70

File: modules/rds_snapshot_explorer/main.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    return summary_data
+
+
+def process_instance(pacu, client, instance):
+    waiter = client.get_waiter('db_instance_available')
+    waiter.wait(
+        DBInstanceIdentifier=instance['DBInstanceIdentifier'],
+        WaiterConfig=WAIT_CONFIG,
+    )
+    password = pacu.input('    Set Master Password for current instance: ')
```
[the remark this answers is not in the document]
Comment (SpenGietz) [reply modules/rds_snapshot_explorer/main.py]: Same comment

File: modules/rds__explore_snapshots/main.py [hunk 8]
```diff
@@ -240,4 +242,6 @@ def get_all_region_instances(client, print):
 def summary(data, pacu_main):
     if 'fail' in data:
         return data['fail']
-    return '  {} Instance(s) Copies Launched'.format(data['instances'])
+    out = '  No issues cleaning up temporary data\n'
+    out += '  {} Instance(s) Copies Launched'.format(data['instances'])
```
Comment (SpenGietz) [inline modules/rds__explore_snapshots/main.py]: "1 Instance(s) Copies" doesn't sound quite right, maybe rephrase this a bit

**Review by SpenGietz — CHANGES_REQUESTED** [review]

File: modules/rds__explore_snapshots/main.py (hunk 6 above)
Comment (alexanderinsa) [reply modules/rds__explore_snapshots/main.py]: The delete_instances function will handle errors related to deleting an instance and print out messages accordingly. success should be set to false upon any issues, but attempts to delete the following temporary resources will continue until all attempts have been made. for each resource.

**Review by SpenGietz — APPROVED** [review]

**Pull request merged**